src/js/messaging/reducers/search.js

```diff
@@ -34,6 +35,40 @@ const initialState = {
 
 export default function modals(state = initialState, action) {
   switch (action.type) {
+    case FETCH_FOLDER_SUCCESS: {
+      const { filter } = action.messages.meta;
+      if (!filter) { return state; }
+
+      const params = {
+        dateRange: { ...initialState.params.dateRange },
```

Comment:
I just wanted a deep enough clone of `initialState`. If it were `dateRange: initialState.params.dateRange`, I think setting `params.dateRange.start = newDate` would affect `initialState.dateRange`.